Backpack/CardButton/Classes/CardButtonViewConfigurator.swift

```diff
@@ -0,0 +1,50 @@
+/*
+ * Backpack - Skyscanner's Design System
+ *
+ * Copyright 2018 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// Hold shared logic between BPKSaveCardButton & BPKShareCardButton
+final class CardButtonViewConfigurator {
+
+    // CardButton Required a larger hit area of at least 48px
+    let widthHeight = BPKSpacingLg * 2
+
+    func createContainedBackgroundCircle() -> UIView {
+        let circle = UIView()
+        circle.backgroundColor = BPKColor.textDisabledOnDarkColor.withAlphaComponent(0.5)
+        circle.isUserInteractionEnabled = false
+        circle.translatesAutoresizingMaskIntoConstraints = false
+        return circle
+    }
+
+    func shouldAddButtonBackground(style: BPKCardButtonStyle) -> Bool {
+        switch style {
+        case .contained:
+            return true
+        case .default, .onDark:
+            return false
+        }
+    }
```

Comment:
We can simplify the switch to just use the 'default' case.
```suggestion
        switch style {
        case .contained:
            return true
        default:
            return false
        }
    }
```

Reply:
sure

Reply:
refactored in 2ba7ea4d07104e1ea426819b04a4cdbf510f9599